File: IPaURLResourceUI/IPaURLResourceUI/IPaURLMultipartFormData.m
//
//  IPaURLMultipartFormData.m
//  IPaURLResourceUI
//
//  Created by IPa Chen on 2014/12/25.
//  Copyright (c) 2014年 IPaPa. All rights reserved.
//

#import "IPaURLMultipartFormData.h"

@interface IPaURLMultipartFormData()

@property (nonatomic,strong) NSMutableData *data;
@end
@implementation IPaURLMultipartFormData
{
    
}
@synthesize boundary = _boundary;
- (NSString *)boundary
{
    if (_boundary == nil) {
        _boundary = [[NSProcessInfo processInfo] globallyUniqueString];
    }
    return _boundary;
}
- (NSMutableData *)data
{
    if (_data == nil) {
        _data = [NSMutableData data];
    }
    return _data;
}

- (void)addStringData:(NSString *)string forName:(NSString*)name
{
    [self.data appendData:[[NSString stringWithFormat:@"--%@\r\nContent-Disposition: form-data; name=\"%@\"\r\n\r\n%@\r\n", self.boundary,name,string] dataUsingEncoding:NSUTF8StringEncoding]];
}
- (void)addIntegerData:(NSInteger)intData forName:(NSString*)name
{
    [self addStringData:[NSString stringWithFormat:@"%ld",intData] forName:name];
}
- (void)addFloatData:(CGFloat)floatData forName:(NSString*)name
{
    [self addStringData:[NSString stringWithFormat:@"%f",floatData] forName:name];
}
- (void)addFileData:(NSData*)fileData fileName:(NSString*)fileName MIMEType:(NSString*)MIMEType forName:(NSString*)name
{
    [self.data appendData:[[NSString stringWithFormat:@"--%@\r\nContent-Disposition: form-data; name=\"%@\"; filename=\"%@\"\r\nContent-Type: %@\r\n\r\n", self.boundary,name,fileName,MIMEType ] dataUsingEncoding:NSUTF8StringEncoding]];
    [self.data appendData:fileData];
    [self.data appendData:[[NSString stringWithFormat:@"\r\n"] dataUsingEncoding:NSUTF8StringEncoding]];
}
- (NSData*)createBodyData
{
    [self.data appendData:[[NSString stringWithFormat:@"--%@--\r\n", self.boundary] dataUsingEncoding:NSUTF8StringEncoding]];
    return self.data;
}
@end

File: IPaURLResourceUI/IPaURLResourceUI/IPaURLMultipartFormData.swift
//
//  IPaURLMultipartFormData.swift
//  IPaURLResourceUI
//
//  Created by IPa Chen on 2015/6/14.
//  Copyright (c) 2015年 IPaPa. All rights reserved.
//

import Foundation

open class IPaURLMultipartFormData {
    let boundary:String = ProcessInfo.processInfo.globallyUniqueString
    lazy var data:NSMutableData = NSMutableData()
    open func addStringData(_ string:String,name:String) {
        let dataString = "--\(boundary)\r\nContent-Disposition: form-data; name=\"\(name)\"\r\n\r\n\(string)\r\n"
        data.append(dataString.data(using: String.Encoding.utf8, allowLossyConversion: false)!)
    }
    open func addFileData(_ fileData:Data,fileName:String,MIMEType:String,name:String) {
        let dataString = "--\(boundary)\r\nContent-Disposition: form-data; name=\"\(name)\"; filename=\"\(fileName)\"\r\nContent-Type: \(MIMEType)\r\n\r\n"
        data.append(dataString.data(using: String.Encoding.utf8, allowLossyConversion: false)!)
        data.append(fileData)
        data.append("\r\n".data(using: String.Encoding.utf8, allowLossyConversion: false)!)
    }
    open func endOfBodyData() {
        let dataString = "--\(boundary)--\r\n"
        data.append(dataString.data(using: String.Encoding.utf8, allowLossyConversion: false)!)
        
    }
    
}
